File: services/shop-backend/src/main/java/io/miragon/shop/adapter/inbound/rest/order/CompleteOrderController.java
package io.miragon.shop.adapter.inbound.rest.order;

import io.miragon.shop.application.port.inbound.PlaceOrderUseCase;
import io.miragon.shop.domain.shared.UserId;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import static io.miragon.shop.adapter.inbound.security.SecurityConfig.USER_ID;

@RestController
@RequestMapping("/api/orders")
public class CompleteOrderController {

    private static final Logger log = LoggerFactory.getLogger(CompleteOrderController.class);

    private final PlaceOrderUseCase completeOrderUseCase;

    public CompleteOrderController(PlaceOrderUseCase completeOrderUseCase) {
        this.completeOrderUseCase = completeOrderUseCase;
    }

    @PostMapping
    public ResponseEntity<OrderDto> completeOrder() {
        var userId = new UserId(USER_ID);
        log.info("Completing order for user: {}", userId);
        var order = completeOrderUseCase.placeOrder(userId);
        return ResponseEntity.ok(OrderDto.fromDomain(order));
    }
}

File: services/shop/shop-backend/src/main/kotlin/io/miragon/shop/adapter/inbound/rest/order/CompleteOrderController.kt
package io.miragon.shop.adapter.inbound.rest.order

import io.github.oshai.kotlinlogging.KotlinLogging
import io.miragon.shop.application.port.inbound.PlaceOrderUseCase
import io.miragon.shop.domain.shared.UserId
import org.springframework.http.ResponseEntity
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.security.oauth2.jwt.Jwt
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/orders")
class CompleteOrderController(
    private val completeOrderUseCase: PlaceOrderUseCase
) {

    private val log = KotlinLogging.logger {}

    @PostMapping
    fun completeOrder(@AuthenticationPrincipal jwt: Jwt): ResponseEntity<OrderDto> {
        val userId = UserId(jwt.subject)
        log.info { "Completing order for user: $userId" }
        val order = completeOrderUseCase.placeOrder(userId)
        return ResponseEntity.ok(OrderDto.fromDomain(order))
    }
}